catalog/to-consul/syncer.go

```diff
@@ -238,27 +271,14 @@ func (s *ConsulSyncer) watchService(ctx context.Context, name string) {
 		s.lock.Lock()
 
 		for _, svc := range services {
-			// If we have a namespace set and the key exactly matches this
-			// namespace, then we skip it.
-			if s.Namespace != "" &&
```

Comment:
can you explain why checking the ns annotations isn't necessary anymore?

Reply:
from rebecca:
> with the addition of the allow and deny lists, this functionality has to change. before, it could be assumed that a sync pod was responsible for everything or only a single namespace. that is no longer the case, since now they are responsible for some arbitrary set of namespaces. because of this, the annotation no longer provides enough information to determine whether the sync pod should delete a specific service or not